pkg/dequeuer/dequeuer.go

```diff
@@ -96,12 +98,37 @@ func (d *SQSDequeuer) ReceiveMessage() (*sqs.Message, error) {
 }
 
 func (d *SQSDequeuer) Start(messageHandler MessageHandler, readinessProbeFunc func() bool) error {
+	numWorkers := math.MaxInt(d.config.Workers, 1)
+
+	d.log.Infof("Starting %d workers", numWorkers)
+	errCh := make(chan error)
+	doneChs := make([]chan struct{}, d.config.Workers)
+	for i := 0; i < numWorkers; i++ {
+		doneChs[i] = make(chan struct{})
+		go func(i int) {
+			errCh <- d.worker(messageHandler, readinessProbeFunc, doneChs[i])
+		}(i)
+	}
+
+	select {
+	case err := <-errCh:
+		return err
```

Comment:
Should we send `doneCh <- struct{}{}` for each of the `doneChs` in this case (like below), or is it not necessary since we are returning `err` here which will stop the entire program?

Reply:
this should be fine

Reply:
This abruptly terminates the goroutines but I think that should be fine for now.

If we wanted to handle the error more gracefully, we can use sync.WaitGroup.